Workspace integration test TestBaseImageBuild and TestParallelBaseImageBuild failed again

### Bug description

The workspace integration tests TestBaseImageBuild and TestParallelBaseImageBuild failed again but with different error message
```shell
running test: /tests/image-builder.test
 === RUN   TestBaseImageBuild
 === RUN   TestBaseImageBuild/database
 === RUN   TestBaseImageBuild/database/it_should_build_a_base_image
     builder_test.go:86: build output: containers are being created
     builder_test.go:86: build output: containers are being created
     builder_test.go:84: image build failed: pod imagebuild-76d339cd-4f00-4ff1-9941-e450ce74a61e has no owner
 --- FAIL: TestBaseImageBuild (2.07s)
     --- FAIL: TestBaseImageBuild/database (2.07s)
         --- FAIL: TestBaseImageBuild/database/it_should_build_a_base_image (2.07s)
 === RUN   TestParallelBaseImageBuild
 === RUN   TestParallelBaseImageBuild/image-builder
 === RUN   TestParallelBaseImageBuild/image-builder/it_should_allow_parallel_build_of_images
     builder_test.go:194: image builder error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
 --- FAIL: TestParallelBaseImageBuild (300.00s)
     --- FAIL: TestParallelBaseImageBuild/image-builder (300.00s)
         --- FAIL: TestParallelBaseImageBuild/image-builder/it_should_allow_parallel_build_of_images (300.00s)
 FAIL
 Test failed at Mon May  2 01:06:39 UTC 2022
 Test suite ended with failure at Mon May  2 01:06:39 UTC 2022
```

### Steps to reproduce

```shell
./dev/preview/install-k3s-kubeconfig.sh
cd test
go test -v ./... \
  -kubeconfig=/home/gitpod/.kube/config \
  -namespace=default \
  -run=TestBaseImageBuild

go test -v ./... \
  -kubeconfig=/home/gitpod/.kube/config \
  -namespace=default \
  -run=TestParallelBaseImageBuild

```

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

--- a/components/ws-manager/pkg/manager/manager.go
+++ b/components/ws-manager/pkg/manager/manager.go
@@ -166,6 +166,8 @@ func (m *Manager) StartWorkspace(ctx context.Context, req *api.StartWorkspaceReq
 	case api.WorkspaceType_IMAGEBUILD:
 		wss, err := m.GetWorkspaces(ctx, &api.GetWorkspacesRequest{
 			MustMatch: &api.MetadataFilter{
+				Owner:       req.Metadata.Owner,
+				MetaId:      req.Metadata.MetaId,
 				Annotations: req.Metadata.Annotations,
 			},
 		})

